internal/session/handle.go

```diff
@@ -32,19 +31,20 @@ func (s *Session) handleOther(
 	}()
 
 	s.handleWG.Go(func() {
-		labels := pprof.Labels("go", "handleOther()", "SessionID", strconv.Itoa(s.sessionID))
-		pprof.Do(ctx, labels, func(_ context.Context) {
+		logging.DoAnnotated(state.NewStateContext(ctx, s.state), func(ctx context.Context) {
 			defer close(resCh)
 
-			ctx := state.NewStateContext(ctx, s.state)
-
 			if err := s.handleCommand(ctx, tag, cmd, resCh, profiler); err != nil {
 				if res, ok := response.FromError(err); ok {
 					resCh <- res
 				} else {
 					resCh <- response.No(tag).WithError(err)
 				}
 			}
+		}, logging.Labels{
+			"Action":    "Handling IMAP command",
+			"SessionID": s.sessionID,
+			"Command":   cmd,
```

Comment:
Potential data/security issue here. Maybe we should just try to get the name of the command rather than the full object?

Reply:
Oops yes i actually intended to just get the commands name but couldn't figure it out so left this as a placeholder but forgot about it. Good catch. Will see if i can get the command name, otherwise will remove

Reply:
Oops yes i actually intended to just get the commands name but couldn't figure it out so left this as a placeholder but forgot about it. Good catch. Will see if i can get the command name, otherwise will remove

Reply:
Tried to get command name but go doesn't make it easy, so removed this line. If need be, can add back later with proper fix.